spynnaker/pyNN/extra_algorithms/neuron_expander.py

```diff
@@ -0,0 +1,107 @@
+# Copyright (c) 2017-2019 The University of Manchester
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+import logging
+from spinn_utilities.log import FormatAdapter
+from spinn_utilities.progress_bar import ProgressBar
+from spinn_front_end_common.utilities.system_control_logic import \
+    run_system_application
+from spinn_front_end_common.utilities.utility_objs import ExecutableType
+from spinnman.model import ExecutableTargets
+from spinnman.model.enums import CPUState
+from spynnaker.pyNN.data import SpynnakerDataView
+from spynnaker.pyNN.models.abstract_models import (
+    AbstractNeuronExpandable, NEURON_EXPANDER_APLX)
+from spinn_front_end_common.utilities.helpful_functions import (
+    write_address_to_user1)
+from spinn_utilities.config_holder import get_config_bool
+
+logger = FormatAdapter(logging.getLogger(__name__))
+
+
+def neuron_expander():
+    """ Run the neuron expander.
+
+    .. note::
+        Needs to be done after data has been loaded.
+    """
+    neuron_bin = SpynnakerDataView.get_executable_path(NEURON_EXPANDER_APLX)
+
+    # Find the places where the neuron expander should run
+    expander_cores, expanded_pop_vertices = _plan_expansion(neuron_bin)
+
+    progress = ProgressBar(expander_cores.total_processors,
+                           "Expanding Neuron Data")
```

Comment:
Does this take long enough that you need a progress bar?

if not use SpynnakerDataView.iterate_placements_by_vertex_type(AbstractNeuronExpandable)

Reply:
It is hard to say at present.  It is possible that the checking of all the cores could be slow on a big machine for example...